internal/controller/controller.go

```diff
@@ -507,18 +509,68 @@ func (c *Controller) deployConfiguration(config *dynamic.Configuration) error {
 		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
 	}
 
-	for _, pod := range podList.Items {
-		log.Debugf("Deploying to pod %s with IP %s", pod.Name, pod.Status.PodIP)
+	if err := deployToPods(podList.Items, config); err != nil {
+		return fmt.Errorf("error deploying configuration: %v", err)
+	}
 
-		if deployErr := c.deployToPod(pod.Name, pod.Status.PodIP, config); deployErr != nil {
-			return fmt.Errorf("error deploying configuration: %v", deployErr)
+	return nil
+}
+
+// deployConfigurationToUnreadyNodes deploys the configuration to the mesh pods.
+func (c *Controller) deployConfigurationToUnreadyNodes(config *dynamic.Configuration) error {
+	podList, err := c.clients.ListPodWithOptions(c.meshNamespace, metav1.ListOptions{
+		LabelSelector: "component==maesh-mesh",
+	})
+	if err != nil {
+		return fmt.Errorf("unable to retrieve pod list: %v", err)
+	}
+
+	if len(podList.Items) == 0 {
+		return fmt.Errorf("unable to find any active mesh pods to deploy config : %+v", config)
+	}
+
+	var unreadyPods []corev1.Pod
+
+	for _, pod := range podList.Items {
+		for _, status := range pod.Status.ContainerStatuses {
+			if !status.Ready {
+				unreadyPods = append(unreadyPods, pod)
+				break
+			}
 		}
 	}
 
+	if err := deployToPods(unreadyPods, config); err != nil {
+		return fmt.Errorf("error deploying configuration: %v", err)
+	}
+
 	return nil
 }
 
-func (c *Controller) deployToPod(name, ip string, config *dynamic.Configuration) error {
+func deployToPods(pods []corev1.Pod, config *dynamic.Configuration) error {
+	var errg errgroup.Group
+
+	for _, p := range pods {
+		pod := p
```

Comment:
Why this assignment?

Reply:
`p` is a variable shared across mutliple goroutines by closure.

However, the p variable is assigned at each iteration. Which means that when the goroutines accessing the variable will start to execute and iterate over them, they will only get the last assigned value, in other words the last item of the slice.

Here is a simple example demonstrating the problem https://play.golang.org/p/fpmNrJ4MMNn